File: sdk/opencode-4-py/src/opencode_4_py/models/agent.py
"""Agent models."""

from __future__ import annotations

from pydantic import BaseModel, Field
from typing import Optional, List, Literal, Dict, Any, Union


class AgentPermission(BaseModel):
    """Agent permission rule."""
    permission: str
    action: Literal["allow", "deny", "ask"]
    pattern: str


PermissionRuleset = List[AgentPermission]


class AgentModel(BaseModel):
    """Agent model reference."""
    provider_id: str = Field(..., alias="providerID")
    model_id: str = Field(..., alias="modelID")
    
    class Config:
        populate_by_name = True


class Agent(BaseModel):
    """AI Agent information."""
    name: str
    description: Optional[str] = None
    mode: Literal["subagent", "primary", "all"]
    native: Optional[bool] = None
    hidden: Optional[bool] = None
    model: Optional[AgentModel] = None
    variant: Optional[str] = None
    prompt: Optional[str] = None
    options: Dict[str, Any] = Field(default_factory=dict)
    permission: Optional[PermissionRuleset] = None
    temperature: Optional[float] = Field(None, alias="temperature")
    top_p: Optional[float] = Field(None, alias="topP")
    color: Optional[str] = None
    steps: Optional[int] = None
    
    class Config:
        populate_by_name = True

File: sdk/oc4j/src/main/java/ai/opencode/sdk/model/agent/Agent.java
package ai.opencode.sdk.model.agent;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Agent {
    private String name;
    private String description;
    private AgentMode mode;
    private Boolean native_;
    
    @JsonProperty("native")
    public void setNative(Boolean native_) {
        this.native_ = native_;
    }
    
    @JsonProperty("native")
    public Boolean getNative() {
        return native_;
    }
    
    private Boolean hidden;
    
    @JsonProperty("topP")
    private Double topP;
    
    private Double temperature;
    private String color;
    private Map<String, Object> permission;
    private Map<String, Object> model;
    private String variant;
    private String prompt;
    private Map<String, Object> options;
    private Integer steps;

    public enum AgentMode {
        @JsonProperty("subagent")
        SUBAGENT,
        @JsonProperty("primary")
        PRIMARY,
        @JsonProperty("all")
        ALL
    }
}